unittests/APITests/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+set (TESTS
+  ReentrantMutex)
+
+list(APPEND LIBS FEXCore)
+
+foreach(API_TEST ${TESTS})
+  add_executable(${API_TEST} ${API_TEST}.cpp)
+  target_link_libraries(${API_TEST} PRIVATE ${LIBS} Catch2::Catch2WithMain)
+
+  catch_discover_tests(${API_TEST}
+    TEST_SUFFIX ".APITest")
+endforeach()
+
+execute_process(COMMAND "nproc" OUTPUT_VARIABLE CORES)
+string(STRIP ${CORES} CORES)
+
+add_custom_target(
```

Comment:
Won't `add_custom_target` cause this to always be executed as part of the build? Is this custom target useful in the first place as opposed to just using `ctest` manually?

Reply:
This follows how all the other unittests are set up.
Each one is set up with a custom target so we can do something like `ninja api_tests` to easily run a series of unit tests.
This is also not always executed as part of the default build. If we wanted that then there would be an `ALL` added to the custom_target.

Reply:
Ah, I misread the CMake documentation about this. The target is "always considered out of date", but I had skimmed over the part that says "nothing depends on the custom target" unless dependencies are explicitly defined.

Thanks for the clarification! I added some tasks to the thunking PR to make sure I don't forget adopting a similar approach for its tests.